client/checkout/utils/platform-checkout.js

```diff
@@ -6,13 +13,69 @@ export const handlePlatformCheckoutEmailInput = ( field ) => {
 	const parentDiv = platformCheckoutEmailInput.parentNode;
 	spinner.classList.add( 'wc-block-components-spinner' );
 
-	const platformCheckoutLocateUser = () => {
+	const iframeWrapper = document.createElement( 'div' );
+	iframeWrapper.setAttribute( 'role', 'dialog' );
+	iframeWrapper.setAttribute( 'aria-modal', 'true' );
+	iframeWrapper.classList.add( 'platform-checkout-sms-otp-iframe-wrapper' );
+	const iframe = document.createElement( 'iframe' );
+	iframe.title = __(
+		'Platform checkout SMS code verification',
+		'woocommerce-payments'
+	);
+	iframe.classList.add( 'platform-checkout-sms-otp-iframe' );
+	iframe.addEventListener( 'load', () => {
+		iframe.classList.add( 'open' );
+	} );
+	iframeWrapper.insertBefore( iframe, null );
+
+	const closeIframe = () => {
+		iframeWrapper.remove();
+		iframe.classList.remove( 'open' );
+		platformCheckoutEmailInput.focus();
+	};
+	iframeWrapper.addEventListener( 'click', closeIframe );
+	window.addEventListener( 'message', ( e ) => {
+		if ( getConfig( 'platformCheckoutHost' ) !== e.origin ) {
+			return;
+		}
+
+		if ( 'close_modal' === e.data.action ) {
+			closeIframe();
+		}
+	} );
```

Comment:
If we add `overflow: hidden` to `body` in `openIframe`, we'll need to also remove it here (or remove the class - whichever method you choose).